core/src/test/java/org/mskcc/oncotree/utils/VersionUtilTest.java

```diff
@@ -0,0 +1,271 @@
+            if (!testValuesMatch(expectedGraphURI, returnedGraphURI)) {
+                failureReport.append(makeMismatchMessage(testVersionName, "graphURI", returnedGraphURI, expectedGraphURI));
+                failureCount = failureCount + 1;
+            }
+            String expectedDescription = expectedVersion.getDescription();
+            String returnedDescription = returnedVersion.getDescription();
+            if (!testValuesMatch(expectedDescription, returnedDescription)) {
+                failureReport.append(makeMismatchMessage(testVersionName, "description", returnedDescription, expectedDescription));
+                failureCount = failureCount + 1;
+            }
+        }
+        if (failureCount > 0) {
+            fail(Integer.toString(failureCount) + " failed test conditions. Details follow... " + failureReport.toString());
+        }
+        return;
+    }
+
+    @Test(expected = InvalidVersionException.class)
+    public void testUnmatchedVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = "";
+        for (String versionName : expectedVersionMap.keySet()) {
+            testVersionName = testVersionName + expectedVersionMap.get(versionName).getVersion();
+        }
+        Version returnedVersion = versionUtil.getVersion(testVersionName);
+        fail("expected exception (InvalidVersionException) did not get thrown");
+    }
+
+    @Test(expected = InvalidVersionException.class)
+    public void testNullVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = null;
+        Version returnedVersion = versionUtil.getVersion(testVersionName);
+        fail("expected exception (InvalidVersionException) did not get thrown");
+    }
+
+    @Test(expected = InvalidVersionException.class)
+    public void testEmptyVersionQuery() throws Exception {
+        setupForTest();
+        String testVersionName = "";
+        Version returnedVersion = versionUtil.getVersion(testVersionName);
+        fail("expected exception (InvalidVersionException) did not get thrown");
+    }
+
+    @Test
+    public void testGetDefaultVersion() throws Exception {
+        setupForTest();
+        Version returnedVersion = versionUtil.getDefaultVersion();
+        if (returnedVersion == null || returnedVersion.getVersion() == null || returnedVersion.getVersion().trim().length() < 1) {
+            fail("returned value from getDefaultVersion() was empty");
+        }
+        if (!returnedVersion.getVersion().equals(VersionUtil.DEFAULT_VERSION)) {
+            fail("returned value from getDefaultVersion() did have the correct version identifier");
+        }
+    }
+
+    private static String[] getRawTestValueSource() {
+        String[] rawTestValueSource = {
+                "oncotree_latest_stable", "urn:x-evn-master:oncotree_2017_06_21", "This is an alias for whatever OncoTree version is the latest stable (timestamped) release.",
```

Comment:
Should this maybe get modeled instead of having arrays of strings that have no clear start/end for a specific version? The parsing code is also a little confusing at first glance.